README.md

```diff
@@ -27,108 +28,132 @@ const schema = {
 1. `@fastify/env` is registered first.
-2. Await the plugin registration or await after()
+2. Await the plugin registration or use `await fastify.ready()`.
 
 ```js
-await fastify.register(fastifyEnv)
+await fastify.register(fastifyEnv);
 // fastify.config can be used in here
 ```
 
 OR
+
 ```js
-fastify.register(fastifyEnv)
-await fastify
+fastify.register(fastifyEnv);
+await fastify.ready();
 // fastify.config can be used in here
 ```
-**NB** Support for additional properties in the schema is disabled for this plugin, with the `additionalProperties` flag set to `false` internally.
 
-### Typescript
-In order to have typing for the fastify instance, you should either:
+**NB:** Support for additional properties in the schema is disabled for this plugin, with the `additionalProperties` flag set to `false` internally.
+
+### TypeScript
 
-- use the `declaration merging` technique to enhance the `FastifyInstance` type with the property and its keys you have defined in the options:
+To properly type the Fastify instance with TypeScript, follow one of these approaches:
+
+#### 1. Declaration Merging
+
+Use declaration merging to enhance the `FastifyInstance` type with the properties and keys defined in your options:
 
 ```typescript
 declare module 'fastify' {
   interface FastifyInstance {
-    config: { // this should be same as the confKey in options
+    config: { // this should be the same as the confKey in options
       // specify your typing here
-      FOO: string
+      FOO: string;
     };
   }
 }
 
-const fastify = Fastify()
-fastify.register(fastifyEnv)
+const fastify = Fastify();
+fastify.register(fastifyEnv);
 
-fastify.config.FOO // will be a string
-fastify.config.BAR // error: Property BAR does not exist on type { FOO: string }
+fastify.config.FOO; // will be a string
+fastify.config.BAR; // error: Property BAR does not exist on type { FOO: string }
 ```
 
-- use the generic function `getEnvs()` to get the already typed object:
+#### 2. Using `getEnvs()` Generic Function
+
+Use the generic function `getEnvs()` to obtain a typed object:
 
 ```typescript
 type Envs = {
-  FOO: string
-}
+  FOO: string;
+};
+
+const fastify = Fastify();
+await fastify.register(fastifyEnv);
+
+const envs = fastify.getEnvs<Envs>(); // envs will be of type Envs
 
-const fastify = Fastify()
-await fastify.register(fastifyEnv)
+envs.FOO; // will be a string
+envs.BAR; // error: Property BAR does not exist on type Envs
+```
+
+To keep types synchronized with the schema, consider using [json-schema-to-ts](https://github.com/ThomasAribart/json-schema-to-ts).
+
+### Handling `.env` Files in TypeScript Projects
 
-const envs = fastify.getEnvs<Envs>() // envs will be of type Envs
+To ensure that the `.env` file is copied to the `dist/` directory after TypeScript compilation, you can add a post-build step to your `package.json` scripts:
```

Comment:
It is very depending on project setup, I use TypeScript and I never copy the `.env` to `dist` file.

Reply:
The problem isn't related to TypeScript, but when you specify a path. 
You should follow what you specify.

I think a note or warning to remind is a good idea, but a full code snippets is not.
People tend to copy and paste the provided code without thinking the reason behind.

Reply:
I agree , will make the necessary changes .

Reply:
I would say something like

> When you pointed to a specify path for your environment file, please aware your environment file should exist in the correct place. Especially in TypeScript project, the `__dirname` and `import.meta.dirname` may not be the same between development and production.